### PR TITLE
Fix/events new

### PR DESCRIPTION
**Pull request by abdullahjankhan1**
## Description

Closes: [PS-293](https://pylonstech.atlassian.net/browse/PS-293)

New endpoint was created to query transaction history of a user
@Params: 
1. User's Address
2. limit
3. offset 
4. denomination (if any)
limit and offset are defaulted to cosmos sdk default values.
@Returns:
List of history for the user, i.e. send, receive, nft buy, nft sell  
---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [x] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [x] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/pylons/keeper/msg_get_tx_history.go [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+
+	// 2. querying cosmos sdk service to get transfer.recipient event
+	// from this event we get 1 types of TxHistory i.e. RECEIVE
+	history, err = txService.GetTxsEvent(context.Background(), &tx.GetTxsEventRequest{Events: []string{
+		types.TransferRecipientEvent + address,
+	}})
+	if err != nil {
+		return nil, err
+	}
+	// helper function to extract specific event information i.e. tranfer
```
Comment (mustafapylons) [inline x/pylons/keeper/msg_get_tx_history.go:107]: Typo in comments "transfer" instead of "tranfer"

File: x/pylons/keeper/msg_get_tx_history.go [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+						}
+					}
+					// case create_item event found extracyt amount at which nft is bought
+					// its identification id's and owner address
+				case types.CreateItemKey:
+					entry.Address = ""
+					for _, attr := range e.GetAttributes() {
+						switch attr.Key {
+						case types.KeyAmount:
+							if len(attr.Value) == types.MinVal {
```
Comment (mustafapylons) [inline x/pylons/keeper/msg_get_tx_history.go:201]: this if/else is multiple places within different functions, code reusability would be better.

File: x/pylons/keeper/msg_get_tx_history_test.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	}
+	userHistory := keeper.QueryEventSender(block)
+	suite.Require().Equal(1, len(userHistory))
+	userHistory = keeper.QueryEventRecipientBank(block)
+	suite.Require().Equal(1, len(userHistory))
+	userHistory = keeper.QueryEventNFTSell(block)
+	suite.Require().Equal(1, len(userHistory))
+}
+
+func (suite *IntegrationTestSuite) TestGetTxHistoryNoHitoryNFT() {
```
Comment (mustafapylons) [inline x/pylons/keeper/msg_get_tx_history_test.go:70]: Probably the intention here was to name the function "TestGetTxHistoryNoHistoryNFT", not missing "s"

**Review by mustafapylons — COMMENTED** [review]
couple typo issues and some code reusability has to be fixed, otherwise code is good.

**Review by mustafapylons — APPROVED** [review]
Looks good to me after the changes.

**Review by MikeSofaer — APPROVED** [review]

**Pull request merged**